inform7/extensions/standard_rules/Sections/Variables and Rulebooks.w

```diff
@@ -520,37 +510,28 @@ Dining Room for three turns". It's as if the player teleports into an
 already-existing world, like some Star Trek crewman, just in time for the
 first command.
 
-(e) All items begin unmentioned, as might be expected.
-
-(f) And the "position player in model world rule" completes the initial
-construction of the spatial model world.
-
-(g) The "start in the correct scenes rule" ensures that we start out
+(f) The "start in the correct scenes rule" ensures that we start out
 in the correct scenes. (This can't wait, because it's just conceivable
 that somebody has written a rule with a preamble like "When play
 begins during the Hunting Season...": it's also where the scene
 Entire Game begins.) That completes the necessary preliminaries before
 ordinary I7 rules can be run.
 
 =
-The start in the correct scenes rule is listed first in the startup rulebook. [7th.]
-The position player in model world rule is listed first in the startup rulebook. [6th.]
-This is the declare everything initially unmentioned rule:
-	repeat with item running through things:
-		now the item is not mentioned.
-The declare everything initially unmentioned rule is listed first in the startup rulebook. [5th]
-The update chronological records rule is listed first in the startup rulebook. [4th.]
-The seed random number generator rule is listed first in the startup rulebook. [3rd.]
-The virtual machine startup rule is listed first in the startup rulebook. [2nd.]
-The initialise memory rule is listed first in the startup rulebook. [1st.]
+The initialise situation rule is listed in the before starting the virtual machine rules.
+The initialise situation rule translates into Inter as "INITIALISE_SITUATION_R".
+
+The initial whitespace rule is listed first in the after starting the virtual machine rules.
+The initial whitespace rule translates into Inter as "INITIAL_WHITESPACE_R".
+
+The position player in model world rule is listed in the after starting the virtual machine rules.
+The position player in model world rule translates into Inter as "POSITION_PLAYER_IN_MODEL_R".
 
-The virtual machine startup rule is defined by Inter as "VIRTUAL_MACHINE_STARTUP_R".
-The initialise memory rule is defined by Inter as "INITIALISE_MEMORY_R".
-The seed random number generator rule is defined by Inter as "SEED_RANDOM_NUMBER_GENERATOR_R".
-The update chronological records rule is defined by Inter as "UPDATE_CHRONOLOGICAL_RECORDS_R".
-The position player in model world rule is defined by Inter as "POSITION_PLAYER_IN_MODEL_R".
+The prepare chronological records rule is listed in the after starting the virtual machine rules.
+The prepare chronological records rule translates into Inter as "UPDATE_CHRONOLOGICAL_RECORDS_R".
```

Comment:
So this is a minor problem: Inform won't let you put the one rule in both an activity and a normal rulebook:

> You've asked to put the rule 'update chronological records rule' into
    the rulebook 'Turn sequence' (the Standard Rules, line 549), which is based
    on an action name and produces nothing; but it was originally written to go
    into a rulebook of a different kind, 'after starting the virtual machine' (source
    text, line 1), which is based on an object and produces nothing. Because
    those kinds are different, 'update chronological records rule' can't go
    into 'Turn sequence' (the Standard Rules, line 549).

So I came up with this solution: create two I7 rules for the one I6 function.

It works, but now test cases are failing because it only gives the name of the first function being called:

```
        if ((R == UPDATE_CHRONOLOGICAL_RECORDS_R)) {
            print "prepare chronological records rule";
            rtrue;
        }
        if ((R == UPDATE_CHRONOLOGICAL_RECORDS_R)) {
            print "update chronological records rule";
            rtrue;
        }
```

So it would probably be better to go back to one rule, but that would mean we'd need to rulebook bases match. I'm pretty sure the starting the virtual machine activity could have any basis king, it shouldn't matter to any of the rules. Does the turn sequence rules actually need to be based on an action name? If it doesn't maybe we could just change both to have a nothing basis?

Reply:
Well I tried changing the basis, and now it says this...

> Before reading 'Starting the virtual machine is an activity on nothing' (Basic
    Inform, line 160), I already knew that 'Starting the virtual machine' is an
    activity on objects, and it is too late to change now.
    Because of this problem, the source could not be translated into a working
    game. (Correct the source text to remove the difficulty and click on Go
    once again.)

(Also, it seems like you can't specify both an activity basis and a `(documented at ...)`.)

Reply:
I've temporarily resolved this with the following, though I don't think it should be the permanent solution:

```
The prepare chronological records rule is listed in the after starting the virtual machine rules.
The prepare chronological records rule translates into Inter as "PREPARE_CHRONOLOGICAL_RECORDS_R".

Include (-
! Awkward hack to get around rulebook basis issues that mean we can't simply include the one rule everywhere we need to
[ PREPARE_CHRONOLOGICAL_RECORDS_R;
	UPDATE_CHRONOLOGICAL_RECORDS_R();
];
-).
```